horovod/torch/__init__.py

```diff
@@ -190,7 +204,10 @@ def broadcast_optimizer_state(optimizer, root_rank):
         for group in optimizer.param_groups:
             for p in group['params']:
                 p.grad = p.data.new(p.size()).zero_()
-        optimizer.step()
+        if type(optimizer).__module__.startswith('horovod'):
```

Comment:
That kind of burns my eyes... I guess we could make it a bit more robust though: 
```
if optimizer.__module__ == DistributedOptimizer.__module__:
```